app/scripts/modules/core/src/application/applicationSearchResultType.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+
+import {
+  AccountCell, BasicCell, HrefCell, searchResultTypeRegistry, ISearchResult,
+  HeaderCell, TableBody, TableHeader, TableRow, SearchResultTab
+} from 'core/search';
+
+export interface IApplicationSearchResult extends ISearchResult {
+  accounts: string[];
+  application: string;
+  cloudProviders: string;
+  createTs: string;
+  description: string;
+  email: string;
+  group: string;
+  lastModifiedBy: string;
+  legacyUdf: boolean;
+  name: string;
+  owner: string;
+  pdApiKey: string;
+  updateTs: string;
+  url: string;
+  user: string;
+}
+
+const cols = {
+  APPLICATION: { key: 'application', label: 'Name', cellRenderer: HrefCell },
+  ACCOUNT: { key: 'accounts', label: 'Account', cellRenderer: AccountCell },
+  EMAIL: { key: 'email', label: 'Owner Email', cellRenderer: BasicCell },
+};
+
+const iconClass = 'fa fa-window-maximize';
+const displayName = 'Applications';
+
+const itemKeyFn = (item: IApplicationSearchResult) => item.application;
+const itemSortFn = (a: IApplicationSearchResult, b: IApplicationSearchResult) =>
+  a.application.localeCompare(b.application);
+
+searchResultTypeRegistry.register({
+  id: 'applications',
+  order: 1,
+  displayName,
+  iconClass,
+  displayFormatter: (searchResult: IApplicationSearchResult) => searchResult.application,
+  components: {
+    SearchResultTab: ({ ...props }) => (
+      <SearchResultTab {...props} iconClass={iconClass} label={displayName} />
+    ),
+
+    SearchResultsHeader: () => (
+      <TableHeader>
+        <HeaderCell col={cols.APPLICATION}/>
+        <HeaderCell col={cols.ACCOUNT}/>
+        <HeaderCell col={cols.EMAIL}/>
+      </TableHeader>
+    ),
```

Comment:
I chose to have these two components (`SearchResultsHeader` and `SearchResultsData`) render their cells individually, instead of have a different component iterate over a list of columns.  I think this way is clearer, although it adds quite a bit more boilerplate.